modules/gdscript/gdscript_parser.cpp

```diff
@@ -4157,6 +4162,39 @@ bool GDScriptParser::onready_annotation(AnnotationNode *p_annotation, Node *p_ta
 	return true;
 }
 
+bool GDScriptParser::tool_button_annotation(AnnotationNode *p_annotation, Node *p_node, ClassNode *p_class) {
```

Comment:
* We need to check that `@tool_button` annotation is not applied to the same function twice.
* We probably shouldn't allow static functions to be annotated with this annotation, although I think it will work?
* Also note that the function can be overridden in a subclass. If it is also annotated with `@tool_button`, then the property list will have two properties with the same name.

Reply:
> We need to check that @tool_button annotation is not applied to the same function twice.

I'll fix this, woops.

>We probably shouldn't allow static functions to be annotated with this annotation

I disagree with unnecessarily forbidding static functions to be annotated without a motivating reason. They can be used in either a static context or one in regards to an object, it doesn't matter. That's on the user.

> Also note that the function can be overridden in a subclass. If it is also annotated with @tool_button, then the property list will have two properties with the same name.

I'm not sure how to disallow this (or update the previous property with new name/icon information). It kind of needs to be duplicated and the last occurrence be the correct/overridden instance. There's no obvious or easy way that I can see to accomplish this. Do you have suggestions here?